File: src/Server/Server.Domain/Models/Metric.cs
namespace Server.Domain.Models;

public class Metric
{
  public Guid AgentId { get; set; }
  public DateTimeOffset Timestamp { get; set; }
  public double CpuUsage { get; set; }
  public double MemoryUsage { get; set; }
  public double NetworkUsage { get; set; }
  public double DiskUsage { get; set; }
  public double Uptime { get; set; }
}

File: src/Server/Server.Ui/src/types/metric.ts
export interface AgentMetricDto {
  agentId: string;
  timestamp: string; // ISO 8601 format
  cpuUsage: number; // in percentage
  memoryUsage: number; // in percentage
  diskUsage: number; // in percentage
  networkUsage: number; // in KB/s
}
